skore/tests/unit/item/test_matplotlib_figure_item.py

```diff
@@ -0,0 +1,73 @@
+import base64
+import io
+import pickle
+
+import pytest
+from matplotlib.figure import Figure
+from matplotlib.pyplot import subplots
+from matplotlib.testing.compare import compare_images
+from skore.persistence.item import ItemTypeError, MatplotlibFigureItem
+
+
+class FakeFigure(Figure):
+    def savefig(self, stream, *args, **kwargs):
+        stream.write(b"<figure>")
+
+
+class TestMatplotlibFigureItem:
+    @pytest.fixture(autouse=True)
+    def monkeypatch_datetime(self, monkeypatch, MockDatetime):
+        monkeypatch.setattr("skore.persistence.item.item.datetime", MockDatetime)
+
+    def test_factory(self, mock_nowstr, tmp_path):
+        figure, ax = subplots()
+        ax.plot([1, 2, 3, 4], [1, 4, 2, 3])
+
+        item = MatplotlibFigureItem.factory(figure)
+
+        # matplotlib being not consistent (`xlink:href` are different between two calls)
+        # we can't compare figure bytes directly
+
+        figure.savefig(tmp_path / "figure.png")
+        pickle.loads(item.figure_bytes).savefig(tmp_path / "item.png")
+
+        assert not compare_images(tmp_path / "figure.png", tmp_path / "item.png", 0)
```

Comment:
```suggestion
        assert compare_images(tmp_path / "figure.png", tmp_path / "item.png", tol=0) is None
```
I had to read the `compare_images` source code to understand the `not` ^^'
That's a great way to test anyway!

Reply:
Fixed in every file using this function:

```
╰─❯ grep -ri compare_images skore/tests                                                                                                                                                                                                     ─╯
skore/tests/unit/project/test_project.py:    assert compare_images(tmp_path / "figure.png", tmp_path / "item.png", 0) is None
skore/tests/unit/item/test_matplotlib_figure_item.py:        assert compare_images(tmp_path / "figure.png", tmp_path / "item.png", 0) is None
skore/tests/unit/item/test_matplotlib_figure_item.py:            compare_images(tmp_path / "figure.png", tmp_path / "item1.png", 0) is None
skore/tests/unit/item/test_matplotlib_figure_item.py:            compare_images(tmp_path / "figure.png", tmp_path / "item2.png", 0) is None
```